src/language/lexer.rs

```diff
@@ -2,26 +2,26 @@
 use std::str::CharIndices;
 use std::iter::Peekable;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq)]
```

Comment:
Style thing: I typically prefer `#[derive(PartialEq, Debug)]` as it better models the importance of the derivations, generally `Debug` is only used when tests fail whereas `PartialEq` is important in non-test code.